digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/dao/ServicePointBusStopDao.scala

```diff
@@ -0,0 +1,262 @@
+      val propertyValue = r.nextLongOption()
+      val propertyDisplayValue = r.nextStringOption()
+      val property = new PropertyRow(
+        propertyId = propertyId,
+        publicId = propertyPublicId,
+        propertyType = propertyType,
+        propertyRequired = propertyRequired,
+        propertyValue = propertyValue.getOrElse(propertyDisplayValue.getOrElse("")).toString,
+        propertyDisplayValue = propertyDisplayValue.orNull,
+        propertyMaxCharacters = propertyMaxCharacters)
+      val created = new Modification(r.nextTimestampOption().map(new DateTime(_)), r.nextStringOption)
+      val modified = new Modification(r.nextTimestampOption().map(new DateTime(_)), r.nextStringOption)
+
+      ServicePointRow(id, externalId, assetTypeId, point, validFrom, validTo, property, created, modified, municipalityCode = municipalityCode)
+    }
+  }
+
+  def assetRowToProperty(assetRows: Iterable[ServicePointRow]): Seq[Property] = {
+    assetRows.groupBy(_.property.propertyId).map { case (key, rows) =>
+      val row = rows.head
+      Property(
+        id = key,
+        publicId = row.property.publicId,
+        propertyType = row.property.propertyType,
+        required = row.property.propertyRequired,
+        values = rows.map(assetRow =>
+          PropertyValue(
+            assetRow.property.propertyValue,
+            propertyDisplayValueFromAssetRow(assetRow))
+        ).filter(_.propertyDisplayValue.isDefined).toSeq,
+        numCharacterMax = row.property.propertyMaxCharacters)
+    }.toSeq
+  }
+
+  private def propertyDisplayValueFromAssetRow(assetRow: ServicePointRow): Option[String] = {
+    Option(assetRow.property.propertyDisplayValue)
+  }
+
+  def fetchAsset(queryFilter: String => String): Seq[ServicePoint] = {
+    val query = """
+        select a.id, a.external_id, a.asset_type_id,
+        a.valid_from, a.valid_to, geometry, a.municipality_code,
+        p.id, p.public_id, p.property_type, p.required, p.max_value_length, e.value,
+        case
+          when e.name_fi is not null then e.name_fi
+          when tp.value_fi is not null then tp.value_fi
+          else null
+        end as display_value,
+        a.created_date, a.created_by, a.modified_date, a.modified_by
+        from asset a
+          join property p on a.asset_type_id = p.asset_type_id
+          left join single_choice_value s on s.asset_id = a.id and s.property_id = p.id and p.property_type = 'single_choice'
+          left join text_property_value tp on tp.asset_id = a.id and tp.property_id = p.id and (p.property_type = 'text' or p.property_type = 'long_text' or p.property_type = 'read_only_text')
+          left join multiple_choice_value mc on mc.asset_id = a.id and mc.property_id = p.id and (p.property_type = 'multiple_choice' or p.property_type = 'checkbox')
+          left join enumerated_value e on mc.enumerated_value_id = e.id or s.enumerated_value_id = e.id
+        where exists (
+            select '1'
```

Comment:
Any idea if this is really necessary?

Reply:
It is to fetch only the mass service stops so there would be no need to filter after

Reply:
@marquesfm Nothing done here. Leaved it as it is